grader/reject.go

```diff
@@ -125,3 +135,28 @@ func checkSubmissionLimitWindow(window *model.SubmittionLimitWindow,
 
     return nil, nil;
 }
+
+func checkLateAcknowledgment(assignment *model.Assignment, lateAcknowledgment bool) (RejectReason, error) {
+    // if the user acknowledges that they are submitting late, do not reject the submission
+    if lateAcknowledgment {
+        return nil, nil
+    }
+
+    policy := assignment.GetLatePolicy()
+    
+    if policy.Type == model.EmptyPolicy || policy.Type == model.BaselinePolicy {
+        return nil, nil
+    }
+
+    dueDate := assignment.DueDate
+
+    if dueDate.IsZero() {
+        return nil, nil
+    }
+
+    if common.NowTimestamp() >= dueDate {
```

Comment:
I don't like the string comparison here.
This heavily depends on us not changing the timestamp format.
Instead, add a method to timestamp (`After()` maybe).